languageserver/integration/commands.go

```diff
@@ -137,82 +142,55 @@ func (i *FlowIntegration) executeScript(conn protocol.Conn, args ...interface{})
 		return nil, err
 	}
 
-	uri, ok := args[0].(string)
+	_, ok := args[0].(string)
 	if !ok {
 		return nil, fmt.Errorf("invalid URI argument: %#+v", args[0])
 	}
 
-	doc, ok := i.server.GetDocument(protocol.DocumentUri(uri))
-	if !ok {
-		return nil, fmt.Errorf("could not find document for URI %s", uri)
-	}
 
 	rawScriptArguments, ok := args[1].([]interface{})
 	if !ok {
 		return nil, fmt.Errorf("invalid script arguments: %#+v", args[1])
 	}
 
-	// TODO: remove this decoding.
-	// The Flow Go SDK doesn't allow sending a script with raw arguments,
-	// so decode them for now (and then they get re-encoded, but that's negligible)
-
-	scriptArguments := make([]cadence.Value, len(rawScriptArguments))
+	var scriptArguments []string
 	for i, rawScriptArgument := range rawScriptArguments {
-		scriptArgumentString, ok := rawScriptArgument.(string)
+		stringArgument, ok := rawScriptArgument.(string)
 		if !ok {
 			return nil, fmt.Errorf("invalid script argument at index %d: %#+v", i, rawScriptArgument)
 		}
 
-		scriptArgument, err := jsoncdc.Decode([]byte(scriptArgumentString))
-		if err != nil {
-			return nil, fmt.Errorf("invalid script argument at index %d: %w", i, err)
-		}
-		scriptArguments[i] = scriptArgument
+		scriptArguments = append(scriptArguments, strings.TrimSuffix(stringArgument, "\n"))
 	}
 
-	script := []byte(doc.Text)
-	res, err := i.flowClient.ExecuteScriptAtLatestBlock(context.Background(), script, scriptArguments)
-	if err != nil {
+	// Pass arguments back to extension
+	codeArguments := fmt.Sprintf("%s", strings.Join(scriptArguments, ","))
```

Comment:
Why comma separate all the arguments and not just sent `scriptArguments` directly to the client?

Reply:
That's a really good question!
I think I was focused on getting a string on extension side, since I could not completely grasp the concept of how we communicate with it.